runners/core-java/src/main/java/org/apache/beam/runners/core/ReduceFnRunner.java

```diff
@@ -919,8 +920,9 @@ private boolean needToEmit(boolean isEmpty, boolean isFinished, PaneInfo.Timing
       // The pane has elements.
       return true;
     }
-    if (timing == Timing.ON_TIME) {
-      // This is the unique ON_TIME pane.
+    if (timing == Timing.ON_TIME
+        && windowingStrategy.getOnTimeBehavior() == Window.OnTimeBehavior.FIRE_ALWAYS) {
```

Comment:
Yes, it is possible. Example:

 - trigger is `AfterWatermark.pastEndOfWindow().withEarlyFirings(X)`
 - EOW and GC happen in one firing (via no allowed lateness or if the watermark goes past EOW and allowed lateness)
 - If the trigger `X` outputs everything early, then you will have an empty `ON_TIME` pane that is also finished.

This creates an equivalence between `EARLY` and `ON_TIME` (via timing and bundling variations) so a user must always filter for "non-late" panes but never filter for "the" on time pane.